packages/console/src/console-history/SizeUsageUI.tsx

```diff
@@ -0,0 +1,211 @@
+  connection: QueryConnectable;
+  UIParams: { defaultUpdateInterval: number; hoverUpdateInterval: number };
+  bgMonitoring?: boolean;
+
+  // in seconds
+  monitorDuration: number;
+}
+
+const SizeUsageUI = ({
+  connection,
+  UIParams,
+  bgMonitoring = true,
+  monitorDuration,
+}: SIzeUsageUIProps): ReactElement => {
+  const [freeMemory, setFreeMemory] = useState<number>(0);
+  const [maxHeapSize, setMaxHeapSize] = useState<number>(999);
+  const [totalHeapSize, setTotalHeapSize] = useState<number>(0);
+  const [hover, setHover] = useState(false);
+
+  const historyUsage = useRef<{ time: number[]; usage: number[] }>({
+    time: [],
+    usage: [],
+  });
+
+  const { defaultUpdateInterval, hoverUpdateInterval } = UIParams;
+
+  const setUsage = (usage: WorkerHeapInfo) => {
+    setFreeMemory(usage.freeMemory);
+    setMaxHeapSize(usage.maximumHeapSize);
+    setTotalHeapSize(usage.totalHeapSize);
+  };
+
+  useEffect(
+    function setUsageUpdateInterval() {
+      const fetchAndUpdate = async () => {
+        const newUsage = await connection.getWorkerHeapInfo();
+
+        if (bgMonitoring || hover) {
+          const usage = newUsage.freeMemory / newUsage.maximumHeapSize;
+          const time = Date.now();
+
+          const past = historyUsage.current;
+          while (
+            past.time.length !== 0 &&
+            time - past.time[0] > monitorDuration * 1000 * 1.5
+          ) {
+            past.time.shift();
+            past.usage.shift();
+          }
+
+          past.time.push(time);
+          past.usage.push(usage);
+
+          historyUsage.current = past;
+        } else {
+          historyUsage.current = { time: [], usage: [] };
+        }
+
+        setUsage(newUsage);
+      };
+      fetchAndUpdate();
+
+      const updateUsage = setInterval(
+        fetchAndUpdate,
+        hover ? hoverUpdateInterval : defaultUpdateInterval
+      );
+      return () => {
+        clearInterval(updateUsage);
+      };
+    },
+    [
+      hover,
+      hoverUpdateInterval,
+      connection,
+      defaultUpdateInterval,
+      monitorDuration,
+      bgMonitoring,
+    ]
+  );
+
+  const toDecimalPlace = (num: number, dec: number) =>
+    Math.round(num * 10 ** dec) / 10 ** dec;
+
+  const decimalPlace = 2;
+  const GbToByte = 1024 ** 3;
+
+  const freeMemoryGB = toDecimalPlace(freeMemory / GbToByte, decimalPlace);
+  const totalHeapGB = toDecimalPlace(totalHeapSize / GbToByte, decimalPlace);
+  const maxHeapGB = toDecimalPlace(maxHeapSize / GbToByte, decimalPlace);
+  const inUseGB = totalHeapGB - freeMemoryGB;
+
+  const getRow = (text: string, size: string, bottomBorder = false) => (
+    <div
+      className={classNames(`heap-usage-info-row`, {
+        'heading-bottom-border': bottomBorder,
+      })}
+    >
+      <div>
+        <h6
+          style={{
+            fontWeight: 'bold',
+          }}
+        >
+          {text}
+        </h6>
+      </div>
+      <div style={{}}>
+        <h6>{size}</h6>
```

Comment:
I'm not sure if this should be a `h6` tag as well. While our app isn't the best from an accessibility standpoint (and we don't enforce accessibility things), it is not great to put the same tag for different levels of importance in the dom. Visually it's fine, but semantically it's probably wrong.

Is the styling of h6 even different from a normal div or span?